### ISSUE TITLE
always get ssim loss = tensor(1., device='cuda:0')

### ISSUE THREAD

**Issue by sunnyHelen** [Po-Hsun-Su/pytorch-ssim#20]
I use SSIM as a loss, but it seems always = 1. I try to print ssim_map. It's all one. I make sure my two img is not the same. Their L1 distance isn't zero. I don't know where the error is. Could you please help me?
![image](https://user-images.githubusercontent.com/28101548/56014622-62959500-5d39-11e9-97c0-a4298a35a3e5.png)

**Comment by YvetteLaw** [comment]
I have the same error. The results are always 1

**Comment by miikatoi** [comment]
Hi,

I had the same problem, but managed to fix it by forcing data type as float. The problem occurred when the input was Byte.

So in my case the fix was simply: `tensor = torch.from_numpy(np.array(img, dtype="float32"))`
instead of: `tensor = torch.from_numpy(np.array(img))`

**Comment by tlwzzy** [comment]
> Hi,
> 
> I had the same problem, but managed to fix it by forcing data type as float. The problem occurred when the input was Byte.
> 
> So in my case the fix was simply: `tensor = torch.from_numpy(np.array(img, dtype="float32"))`
> instead of: `tensor = torch.from_numpy(np.array(img))`

perfectly solved my problem. THANKS a lot !

**Comment by Nonmy** [comment]
> Hi,
> 
> I had the same problem, but managed to fix it by forcing data type as float. The problem occurred when the input was Byte.
> 
> So in my case the fix was simply: `tensor = torch.from_numpy(np.array(img, dtype="float32"))` instead of: `tensor = torch.from_numpy(np.array(img))`

Thanks！
